dataplane/docs/dataplaneforsonic.md

```diff
@@ -0,0 +1,61 @@
+│  ┌─────────────────────────────────────────────┐  ┌──────────────────┐  │
+│  │                                             │  │                  │  │
+│  │  ┌───────────────────────────────────────┐  │  │                  │  │
+│  │  │                                       │  │  │                  │  │
+│  │  │   ┌───────────────────┐            ┌──┴──┴──┴──┐               │  │
+│  │  │   │                   │            │ Ethernet0 │               │  │
+│  │  │   │ ┌──────────────┐  │            └──┬──┬──┬──┘               │  │
+│  │  │   │ │              │  │               │  │  │                  │  │
+│  │  │   │ │ CPU Port MGR │  │               │  │  │                  │  │
+│  │  │   │ │ SAI Impl  ───┼──┼─────gRPC──────┼──┼──┼───► saiserver    │  │
+│  │  │   │ │              │  │               │  │  │                  │  │
+│  │  │   │ │ libsai.so    │  │               │  │  │                  │  │
+│  │  │   │ └──────────────┘  │               │  │  │                  │  │
+│  │  │   │                   │               │  │  │                  │  │
+│  │  │   │                   │               │  │  │                  │  │
+│  │  │   │ syncd             │               │  │  │                  │  │
+│  │  │   └───────────────────┘            ┌──┴──┴──┴──┐               │  │
+│  │  │                                    │ Ethernet4 │               │  │
+│  │  │                                    └──┬──┬──┬──┘               │  │
+│  │  │ VM                                    │  │  │                  │  │
+│  │  └───────────────────────────────────────┘  │  │                  │  │
+│  │                                             │  │                  │  │
+│  │  SONiC VM Runner                            │  │ Dataplane        │  │
+│  └─────────────────────────────────────────────┘  └──────────────────┘  │
+│                                                                         │
+│                                                                         │
+│                                                    ┌──────┐  ┌──────┐   │
+│ pod                                                │ eth1 │  │ eth2 │   │
+└────────────────────────────────────────────────────┴──────┴──┴──────┴───┘
+```
+
+## libsai
+
+The dataplane libsai is shared object that implements the SAI API. It composed of almost entirely autogenerated code that converts the C structs to protobuf struct. The code is generated by the apigen Go library.
+
+In addition to SAI implementation, the CPU Port manager runs as a another thread in the library. The CPU port manager is responible for handling IPs address assign the interfaces inside the VM.
+
+The process for handling packets destined for the host is:
+
+1. A hostif is created called Ethernet0 with OID 3.
+2. saiserver receives the request and sets up the dataplane
+3. saiserver notifies the CPU manager that a hostif is created
+4. the user assignes the IP 192.168.1.100/24 to the Ethernet0
```

Comment:
Is this 192.168.* address prefix always used? Also, who is the user? how does the user config this address?

Reply:
this is just an example. any IP is fine, user is a person configuring the device